lambdas/service/app.py

```diff
@@ -1598,22 +1686,34 @@ def fetch_file_manifest_with_token(token: str):
 
 def _file_manifest(fetch: bool, token_or_key: Optional[str] = None):
     request = app.current_request
+    is_post = request.method == 'POST'
+    if (
+        is_post
+        and request.headers.get('content-type') == 'application/x-www-form-urlencoded'
+        and request.raw_body != b''
+    ):
+        raise BRE('The body must be empty for a POST request of content-type '
+                  '`application/x-www-form-urlencoded` to this endpoint')
     query_params = request.query_params or {}
     _hoist_parameters(query_params, request)
+    if is_post and 'wait' not in query_params:
```

Comment:
Two lines down a different idiom is used to set the default. Be consistent with precedent.

Are you sure the default is needed at this level? You should avoid the case where a default is injected when the parameter is actually disallowed. I'm a bit confused by the fact that the `wait` parameter is conditional on `is_post` in one place and on `fetch` in another. I don't remember: Did I ask you to allow `wait` in use cases other than the one we actually need it for, i.e., with `curl`?

Reply:
> Are you sure the default is needed at this level? You should avoid the case where a default is injected when the parameter is actually disallowed.

Yes, the `wait` parameter only applies if the initial request was POST, in which case we want to set a default (`1`) if the param was not specified.

> I'm a bit confused by the fact that the `wait` parameter is conditional on `is_post` in one place and on `fetch` in another. I don't remember: Did I ask you to allow `wait` in use cases other than the one we actually need it for, i.e., with `curl`?

If an initial POST request returns a 301 with a token, the `wait` parameter is carried over from the initial request, so that each subsequent request to the non-fetch endpoint will have the same `wait` parameter. This is why the param is conditional on `is_post` in one section (when setting a default), and `is not fetch` in another section (when processing a non-initial request). I've added a brief comment to the latter to help clarify.